### ISSUE TITLE
[terra-dev-site] Fails to generate site on windows devices due to windows style file paths

### ISSUE THREAD

**Issue by revoyrm** [cerner/terra-application#280]
# Bug Report

## Description
When developing on windows and running npm start, it outputs errors caused by windows file paths with backslashes instead of forward slashes. The errors are in terra-dev-site\webpack\loaderUtils\generateNavigationConfig.js and can be fixed in executeSearchPatterns and generateNavigationConfig. Below are the 2 errors.

Error 1: 
![image](https://user-images.githubusercontent.com/11396904/142272736-d15e7ed8-eb17-42fa-b72b-188483011aef.png)

When I fix that error in node_modules/@cerner/terra-dev-site/webpack/loaderUtils/generateNavigationConfig.js - executeSearchPatterns, I get the 2nd windows file path error.

Error 2: 
![image](https://user-images.githubusercontent.com/11396904/142273145-a7906460-3efb-4049-8b16-20fad9628b28.png)

Which can be fixed in  terra-dev-site/webpack/loaderUtils/generateNavigationConfig.js - generatePagesConfig.

## Steps to Reproduce

1. Run a site using terra-dev-site 7.0.0 on a windows device

## Additional Context / Screenshots
Here is a link to a conversation in teams about this issue 
https://teams.microsoft.com/l/message/19:3ed4d4d6ffba48138550b9a91a2b50b4@thread.skype/1634129949052?tenantId=fbc493a8-0d24-4454-a815-f4ca58e8c09d&groupId=45f47bfa-38a9-444d-8933-9c20266ae38c&parentMessageId=1634129949052&teamName=Terra&channelName=UX%20Support&createdTime=1634129949052

## Possible Solution
Below is the fix which I've tested on windows and WSL. I have not been able to test it on a mac as I do not have access to one. 

The fix is to run .replace(/\\\\/g, '/') on the following 3 variables:

Error 1: 
entryPoint and filePath on line -> https://github.com/cerner/terra-application/blob/main/packages/terra-dev-site/src/webpack/loaderUtils/generateNavigationConfig.js#L239

Error 2: 
content.filepath on this line -> https://github.com/cerner/terra-application/blob/main/packages/terra-dev-site/src/webpack/loaderUtils/generateNavigationConfig.js#L305

This replace "\\\\" with "/". It should only affect windows file paths and leave linux style file paths alone.

## Environment

* Component Name and Version: EDM/media-viewer-js
* Browser Name and Version: Chrome
* Node/npm Version: node: 12.14.1 - npm: 6.13.4
* Webpack Version: ^4
* Operating System and version (desktop or mobile): Windows 10

## @ Mentions
@benbcai

**Comment by benbcai** [comment]
An internal issue has been logged for this issue.
